### PR TITLE
Add localization

### PR DESCRIPTION
**Pull request by sprucely**
i18next requires a bit of ceremony to get configured, but it seems to be the most popular choice.

### CODE REVIEW

**Review by jSylvestre — APPROVED** [review]
Didn't test it, but looks good to me.

File: HarvestMobile/App.tsx [hunk 1]
```diff
@@ -1,19 +1,31 @@
...
 export default function App() {
   const isLoadingComplete = useCachedResources();
   const colorScheme = useColorScheme();
+  const isI18nInitialized = useState(false);
 
-  if (!isLoadingComplete) {
+  useEffect(() => {
+    i18n.init()
+      .then(() => {
+        isI18nInitialized.set(true);
```
Comment (srkirkland) [inline HarvestMobile/App.tsx:23]: is this a thing now instead of doing `const [isI18nInitialized, setIsI18nInitialized] = useState(false)`?

**Review by srkirkland — APPROVED** [review]

File: HarvestMobile/App.tsx (hunk 1 above)
Comment (sprucely) [reply HarvestMobile/App.tsx:23]: It is for hook-state. Unfortunate that they didn't choose a different name, such as `useHookState`.

Comment (srkirkland) [reply HarvestMobile/App.tsx:23]: oh that sucks

**Pull request merged**